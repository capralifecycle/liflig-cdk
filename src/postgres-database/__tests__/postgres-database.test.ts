import "@aws-cdk/assert/jest"
import { App, Stack } from "@aws-cdk/core"
import * as ec2 from "@aws-cdk/aws-ec2"
import "jest-cdk-snapshot"
import { PostgresDatabase } from ".."

test("create postgres database", () => {
  const app = new App()
  const supportStack = new Stack(app, "SupportStack")
  const stack = new Stack(app, "Stack")

  const vpc = new ec2.Vpc(supportStack, "Vpc")

  new PostgresDatabase(stack, "Database", {
    vpc: vpc,
    instanceType: ec2.InstanceType.of(
      ec2.InstanceClass.BURSTABLE3,
      ec2.InstanceSize.MICRO,
    ),
    instanceIdentifier: "example-database-v1",
    snapshotIdentifier: undefined,
    usePublicSubnets: true,
  })

  expect(stack).toMatchCdkSnapshot()
})
