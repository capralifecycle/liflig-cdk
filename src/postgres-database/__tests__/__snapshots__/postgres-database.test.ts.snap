// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create postgres database 1`] = `
Object {
  "Resources": Object {
    "DatabaseDbE7C02057": Object {
      "DeletionPolicy": "Snapshot",
      "Properties": Object {
        "AllocatedStorage": "25",
        "AllowMajorVersionUpgrade": true,
        "BackupRetentionPeriod": 35,
        "CopyTagsToSnapshot": true,
        "DBInstanceClass": "db.t3.micro",
        "DBInstanceIdentifier": "example-database-v1",
        "DBName": "app",
        "DBSubnetGroupName": Object {
          "Ref": "DatabaseDbSubnetGroupFB62376A",
        },
        "Engine": "postgres",
        "EngineVersion": "12",
        "MasterUserPassword": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{{resolve:secretsmanager:",
              Object {
                "Ref": "DatabaseDbSecret1098DC6E",
              },
              ":SecretString:password::}}",
            ],
          ],
        },
        "MasterUsername": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{{resolve:secretsmanager:",
              Object {
                "Ref": "DatabaseDbSecret1098DC6E",
              },
              ":SecretString:username::}}",
            ],
          ],
        },
        "MultiAZ": true,
        "PubliclyAccessible": false,
        "StorageEncrypted": true,
        "StorageType": "gp2",
        "VPCSecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "DatabaseDbSecurityGroupE6612FBB",
              "GroupId",
            ],
          },
        ],
      },
      "Type": "AWS::RDS::DBInstance",
      "UpdateReplacePolicy": "Snapshot",
    },
    "DatabaseDbSecret1098DC6E": Object {
      "Properties": Object {
        "Description": Object {
          "Fn::Join": Array [
            "",
            Array [
              "Generated by the CDK for stack: ",
              Object {
                "Ref": "AWS::StackName",
              },
            ],
          ],
        },
        "GenerateSecretString": Object {
          "ExcludeCharacters": " %+~\`#$&*()|[]{}:;<>?!'/@\\"\\\\",
          "GenerateStringKey": "password",
          "PasswordLength": 30,
          "SecretStringTemplate": "{\\"username\\":\\"master\\"}",
        },
      },
      "Type": "AWS::SecretsManager::Secret",
    },
    "DatabaseDbSecretAttachment34026B56": Object {
      "Properties": Object {
        "SecretId": Object {
          "Ref": "DatabaseDbSecret1098DC6E",
        },
        "TargetId": Object {
          "Ref": "DatabaseDbE7C02057",
        },
        "TargetType": "AWS::RDS::DBInstance",
      },
      "Type": "AWS::SecretsManager::SecretTargetAttachment",
    },
    "DatabaseDbSecurityGroupE6612FBB": Object {
      "Properties": Object {
        "GroupDescription": "Security group for Db database",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "VpcId": Object {
          "Fn::ImportValue": "SupportStack:ExportsOutputRefVpc8378EB38272D6E3A",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "DatabaseDbSubnetGroupFB62376A": Object {
      "Properties": Object {
        "DBSubnetGroupDescription": "Subnet group for Db database",
        "SubnetIds": Array [
          Object {
            "Fn::ImportValue": "SupportStack:ExportsOutputRefVpcPublicSubnet1Subnet5C2D37C4FFA2B456",
          },
          Object {
            "Fn::ImportValue": "SupportStack:ExportsOutputRefVpcPublicSubnet2Subnet691E08A351552740",
          },
        ],
      },
      "Type": "AWS::RDS::DBSubnetGroup",
    },
  },
}
`;
